docs/adr/adr-0017-finding-accessible-colors-and-creating-design-tokens.md

```diff
@@ -0,0 +1,123 @@
+* People working on this project all are developers,
+* Our components are only for the Web and as close as possible to the standards.
+
+For these reasons, we chose to **only rely on CSS variables and avoid using additional tools** like [Style Dictionary](https://amzn.github.io/style-dictionary/#/) that would only have added more complexity with no actual benefits.
+
+#### Variable naming & architecture
+When creating our variables, we tried to think as much as possible about Developer Experience, theming and customisation.
+
+The [Design Tokens - What are they & how will they help you?](https://lukasoppermann.medium.com/design-tokens-what-are-they-how-will-they-help-you-b73f80f602ab) article helped us a lot to build a system that could scale and allow for customisation later on.
+In this article, the author distinguishes two kind of tokens:
+* "choices" - our colors. For instance:
+```css
+--color-black: #262626;
+```
+* "decisions" - implementations of our colors. For instance:
+```css
+--color-text-default: var(--color-black);
+```
+
+Choices should **never be used directly inside components**. They are to be used as values for **"Decisions" variables** which in turn are to be used inside components.
+
+Creating and using "Decisions" variables has several advantages:
+* If named correctly, **"Decisions" can guide developers and avoid mistakes**. For instance, with `--color-text-default`, the developer knows that this color is to be used for text and not background, and that it represents the default color for text.
+* These token names are not related to their value. **Their name is still valid even if you change their value**, especially when planning for a dark mode. For instance, it does not matter whether `--color-text-default` is black or white, this variable value can be overridden without any problem. Whereas overriding `--color-black` value with a white color value would make the things very confusing for developers.
+* We can **enforce some kind of basic accessibility rules for color usage**. For instance, since most of the lightest shades of our colors are not contrasted enough to be used with our default (white) or neutral (light grey) backgrounds, there is no "Decisions" variable referring to this color for text, only as a background.
+
+## What's next ?
+### Accessibility
+* Texts and icons used in the `cc-tile-instances` component are still legacy and not contrasted enough. This will be handled during the scalability rework (see [issue #407](https://github.com/CleverCloud/clever-components/issues/407)).
+* Focus styles and colors.
+* Icon colors (status of the apps / addons used in the menu and dashboards).
+
+### Theming & customisation
+* Allow for customisation and theming. This means we allow external override of our "Decisions" variables (may need to switch to CSS files for variables instead of JS).
```

Comment:
I would be a bit more precise: instead of CSS inside JavaScript template string with Lit

(something like that)